sre_yield/__init__.py

```diff
@@ -563,8 +563,8 @@ def __init__(self, pattern, flags=0, charset=CHARSET, max_count=None):
             sre_constants.MIN_REPEAT: self.max_repeat_values,
             sre_constants.MAX_REPEAT: self.max_repeat_values,
             sre_constants.AT: self.nothing_added,
-            sre_constants.ASSERT: self.empty_list,
-            sre_constants.ASSERT_NOT: self.empty_list,
+            sre_constants.ASSERT: self.maybe_save,
+            sre_constants.ASSERT_NOT: self.nothing_added,
```

Comment:
If you make `__init__` take some a `relaxed` flag, setting these both to `nothing_added` is good.  You can probably skip `check_anchor_state` when in that mode as well.

Reply:
Ok, a `relax` flag sounds like a good compromise.

Reply:
this PR now redundant as https://github.com/google/sre_yield/pull/21 does the trick.